test/gtest/uct/uct_test.cc

```diff
@@ -58,6 +58,58 @@ uct_test::~uct_test() {
     uct_config_release(m_md_config);
 }
 
+void uct_test::set_interface_rscs(char *md_name, cpu_set_t local_cpus,
+                                  struct ifaddrs *ifa,
+                                  std::vector<resource>& all_resources)
+{
+    int i;
+
+    /* Create two resources on the same interface. the first one will have the
+     * ip of the interface and the second one will have INADDR_ANY */
+    for (i = 0; i < 2; i++) {
+        resource rsc;
+        rsc.md_name    = md_name,
+        rsc.local_cpus = local_cpus,
+        rsc.tl_name    = "sockaddr",
+        rsc.dev_name   = ifa->ifa_name;
+        rsc.dev_type   = UCT_DEVICE_TYPE_NET;
+
+        if (i == 0) {
+            /* first rsc */
+            if (ifa->ifa_addr->sa_family == AF_INET) {
+                memcpy(&rsc.listen_if_addr, ifa->ifa_addr, sizeof(struct sockaddr_in));
```

Comment:
there is a lot of code duplication here, sounds like you want to add function like `init_sockaddr_rsc(&rsc, sockaddr *listen_addr, sockaddr *connect_addr, ...)` and call it once with [ifa->ifa_addr,ifa->ifa_addr] and another time with [ifa->ifa_addr,INADDR_ANY]